akka-http-core/src/main/scala/akka/http/impl/engine/client/pool/NewHostConnectionPool.scala

```diff
@@ -439,8 +441,17 @@ private[client] object NewHostConnectionPool {
 
             emitRequest(newRequest)
           }
-          def close(): Unit = {
-            requestOut.complete()
+
+          /**
+           * If regular is true connection is closed, otherwise, it is aborted.
+           *
+           * A connection should be closed regularly after a request/response with `Connection: close` has been completed.
+           * A connection should be aborted after failures.
+           */
+          def close(regular: Boolean): Unit = {
+            if (regular) requestOut.complete()
+            else requestOut.fail(new IllegalStateException("pool slot shutdown") with NoStackTrace)
```

Comment:
I wonder if `regular: Boolean` should be `error: Option[String]` so we can add a 'cause' to the stacktrace in the message (since we don't have a stack trace here)?

Reply:
Good idea. Done.